tfhe/src/integer/gpu/mod.rs

```diff
@@ -3855,9 +4196,24 @@ pub unsafe fn unchecked_signed_scalar_div_integer_radix_kb_assign_async<
     };
 
     let ct_modulus = numerator.d_blocks.ciphertext_modulus().raw_modulus_float();
-    let ms_noise_reduction_key_ffi =
-        prepare_cuda_ms_noise_reduction_key_ffi(noise_reduction_key, ct_modulus);
-    let allocate_ms_array = noise_reduction_key.is_some();
+    // Initializes as NoReduction and change variables later if otherwise
+    let mut noise_reduction_type = PBSMSNoiseReductionType::NoReduction;
+    let mut ms_noise_reduction_key_ffi = prepare_cuda_ms_noise_reduction_key_ffi(None, ct_modulus);
+
+    match ms_configuration {
```

Comment:
This seems to be repeated many times, would be maybe worth wrapping it in a function?